src/Pact/Types/Gas.hs

```diff
@@ -54,10 +65,15 @@ data GasArgs
   | GUserApp
 
 newtype GasLimit = GasLimit Word64
-  deriving (Eq,Ord,Num,Real,Integral,Enum,Show)
+  deriving (Eq,Ord,Num,Real,Integral,Enum,Show,Serialize,NFData,Generic,ToTerm)
 instance Pretty GasLimit where
   pretty (GasLimit g) = viaShow g
 
+instance ToJSON GasLimit where
+  toJSON (GasLimit d) = toJSON (ParsedInteger (fromIntegral d))
+instance FromJSON GasLimit where
+  parseJSON v = parseJSON v >>= \(ParsedInteger d) -> return (GasLimit (fromIntegral d))
+
 data GasModel = GasModel
   { gasModelName :: Text
   , gasModelDesc :: Text
```

Comment:
There are several places in `TransactionExec` where we vendor in our own "lenses" for both `GasLimit` and `GasPrice`. If we could add some accessors to those newtypes and derive those in Pact, that would be very appreciated.

Reply:
Haha, you'll see -- I have the CW PR ready. You'll never lens in again :)